proto/src/domain/transactions.rs

```diff
@@ -0,0 +1,83 @@
+use std::{
+    collections::BTreeMap,
+    fmt::{Display, Formatter},
+};
+
+use miden_node_utils::formatting::{format_map, format_opt};
+use miden_objects::Digest;
+
+use crate::{
+    domain::accounts::AccountState,
+    errors::{MissingFieldHelper, ParseError},
+    generated::responses::{GetTransactionInputsResponse, NullifierTransactionInputRecord},
+};
+
+// TRANSACTION INPUTS
+// ================================================================================================
+
+/// Information needed from the store to verify a transaction.
+#[derive(Debug)]
+pub struct TransactionInputs {
+    /// The account state in the store corresponding to tx's account ID
+    pub account_state: AccountState,
+
+    /// Maps each consumed notes' nullifier to block number, where the note is consumed
+    /// (`zero` means, that note isn't consumed yet)
+    pub nullifiers: BTreeMap<Digest, u32>,
+}
```

Comment:
Looking at this now, I no longer think it make sense to have the same object shared between the store and the block producer (primarily because it forces us to build a `BTreeMap` in the store, when we really don't need to).

So, I'm thinking we should:
- Move this struct to the block producer crate.
- On the store side, either create a separate struct or build ` GetTransactionInputsResponse` directly.

I would also simplify `TransactionInputs` (and maybe the whole message) to look as follows:
```Rust
pub struct TransactionInputs {
    pub account_id: AccountId,
    pub account_hash: Option<Digest>,
    pub nullifiers: BTreeMap<Digest, u32>,
}
```